src/utils/utils_load_library.h

```diff
@@ -32,29 +39,48 @@ extern "C" {
 
 #ifdef _WIN32 /* Windows */
 
-static inline void *util_open_library(const char *filename) {
+static inline void *util_open_library(const char *filename, int dlopenFlags) {
+    if (dlopenFlags == UTIL_OPEN_LIBRARY_GLOBAL) {
+        // not use for win
+    }
+    (void)dlopenFlags; //unused for win
     return LoadLibrary(TEXT(filename));
 }
 
 static inline int util_close_library(void *handle) {
     // If the FreeLibrary function succeeds, the return value is nonzero.
     // If the FreeLibrary function fails, the return value is zero.
-    return (FreeLibrary(handle) == 0);
+    return (FreeLibrary((HMODULE)handle) == 0);
 }
 
-static inline void *util_get_symbol_addr(void *handle, const char *symbol) {
-    return GetProcAddress(handle, symbol);
+static inline void *util_get_symbol_addr(void *handle, const char *symbol,
+                                         const char *libname) {
+    if (!handle) {
+        if (libname == NULL) {
+            return NULL;
+        }
+        handle = GetModuleHandle(libname);
+    }
+    return GetProcAddress((HMODULE)handle, symbol);
 }
 
 #else /* Linux */
 
-static inline void *util_open_library(const char *filename) {
-    return dlopen(filename, RTLD_LAZY);
+static inline void *util_open_library(const char *filename, int dlopenFlags) {
+    if (dlopenFlags == UTIL_OPEN_LIBRARY_GLOBAL) {
```

Comment:
I know we're passing only one flag for now, but technically `flags` is plural, so you could pass here more than one flag, hence you can't simply check `==`, rather use `&` to check if this precise flag is one of the passed flags.

Also, you'd have to remove your custom flag and replace it with the proper one (`RTLD_GLOBAL` in this case)
You could also use a different variable to translate any of our flags to the proper dlopen flags😉